src/virt.rs

```diff
@@ -497,7 +497,7 @@ impl VirtContext {
         self.csr.mcause = self.trap_info.mcause;
         self.csr.mstatus = self.trap_info.mstatus;
         self.csr.mtval = self.trap_info.mtval;
-        self.csr.mip = self.trap_info.mip;
+        self.csr.mip = self.trap_info.mip & !mie::SEIE_FILTER | self.csr.mip & mie::SEIE_FILTER; // do not take SEIE into account
```

Comment:
Could you add a bit more details about why we do not take it into account?

You can mode it below `self.csr.mepc = ...` to make some space for a multi-line comment if needed. But this is a tricky part, so it needs to be commented with enough detail for someone coming later to understand what is going on.
In particular, it is clear from the code **what** you are doing here, but it is not clear **why** you are doing it. Adding a comment talking briefly about the two SEIE bits and which one we want to keep would help.